instant-xml-macros/src/lib.rs

```diff
@@ -181,13 +185,47 @@ pub fn to_xml(input: TokenStream) -> TokenStream {
 
     serializer.add_footer(&root_name, &mut output);
 
+    let current_prefixes: BTreeSet<&str> = serializer.get_keys_set();
     TokenStream::from(quote!(
         impl ToXml for #ident {
-            fn write_xml<W: ::std::fmt::Write>(&self, write: &mut W) -> Result<(), instant_xml::Error> {
-                write.write_str(&(#output))?;
+            fn write_xml<W: ::std::fmt::Write>(&self, write: &mut W, parent_prefixes: Option<&mut std::collections::BTreeSet<&str>>) -> Result<(), instant_xml::Error> {
```

Comment:
You're not getting line wrapping from rustfmt here because you're inside a macro, but would be good to make sure lines are still wrapped after 100 columns.

Reply:
It is not valid anymore, as I changed the "+" with `push_str`, so it is formatted earlier during macro processing, not inside.

Reply:
In instant-xml-macros we still have all this code inside the `quote!()` macro, so it doesn't get formatted as usual IIRC.

Reply:
Hmm, I did some tests and when I'm expanding the code with `cargo expand` it is formatted correctly. I will dive deeper into this.